.travis.yml

```diff
@@ -49,7 +51,7 @@ install:
 script:
   - jdk_switcher use oraclejdk8
   - ./gradlew classes testClasses -S
-  - jdk_switcher use $TRAVIS_JDK_VERSION
+  - if [[ $TRAVIS_JDK_VERSION == "oraclejdk11" ]]; then export JAVA_HOME=$HOME/oraclejdk11; fi
```

Comment:
What provides this JDK?

Reply:
TravisCI installs it by [install-jdk.sh](https://github.com/sormuras/bach#install-jdksh).

Here is [official documentation about how to switch JDK 10 and up](https://docs.travis-ci.com/user/languages/java/#switching-jdks-to-java-10-and-up-within-one-job).

Reply:
Just wondering that we first switch to jdk 8 but next line check if the jdk version is 11? How is this possible?

Reply:
9 and 10 have already achieved their EOL. Recently Java changed their release policy.

Reply:
I was not wondering where 9 and 10 are, I was trying to understand why the script switches to java 8 but a line after tests if it will be 11? I simply can't understand this part of the script. If I switch to 8 I expect 8 to be used, not 11.

Reply:
By this change, we'll have two build jobs [like this](https://travis-ci.org/spotbugs/spotbugs/builds/448196622).

One will **build and test** spotbugs with JDK8,  
another will **build** spotbugs with JDK8 and **test** it with JDK11.

Its background is:

1. We distribute binary built by JDK8, so build part should always use JDK8 (realized by `jdk_switcher use oraclejdk8` in `.travis.yml`)
2. Target classes to analyse can be compiled by different JDK in users' environment, and JVM that runs SpotBugs also can be different JDK. So test part should have several patterns. (realized by `if [[ $TRAVIS_JDK_VERSION == "oraclejdk11" ]]; then ...` in `.travis.yml`)

In other words, our build matrix is like this:

|#|JDK to compile|JDK to run test|
|-|----|----|
|1|JDK8|JDK8|
|2|JDK8|JDK11|

> If I switch to 8 I expect 8 to be used, not 11.

That's right. For the 1st pattern, we never use JDK11. We use it only for the 2nd pattern.

Reply:
Thanks Kengo. This now makes sense, but I beat next time I will see this code I will ask the same question again :-).